pkg/reconciler/v1alpha1/pipelinerun/pipelinerun_test.go

```diff
@@ -278,6 +282,43 @@ func TestReconcile(t *testing.T) {
 	if _, exists := reconciledRun.Status.TaskRuns["test-pipeline-run-success-unit-test-1"]; exists == false {
 		t.Errorf("Expected PipelineRun status to include TaskRun status")
 	}
+
+	if err := clients.Pipeline.Pipeline().TaskRuns("foo").Delete("test-pipeline-run-success-unit-test-1", &metav1.DeleteOptions{GracePeriodSeconds:new(int64)}); err != nil {
+		t.Fatalf("Failed to delete TaskRun: %s", err)
+	}
```

Comment:
In general I agree - I decided to piggyback on the existing test here because it was already pretty general (`TestReconcile` as the name, and covering the whole reconciliation process), and because it saved copy-pasting a bunch of `Pipeline`/`PipelineRun`/`Task`/etc definitions around. But I'd be happy to switch it up - I'll give that a try shortly.